components/dialog/dialog-confirm.js

```diff
@@ -0,0 +1,58 @@
+import { css, html, LitElement } from 'lit-element/lit-element.js';
+import { DialogMixin } from './dialog-mixin.js';
+import { dialogStyles } from './dialog-styles.js';
+import { getUniqueId } from '../../helpers/uniqueId.js';
+import { heading3Styles } from '../typography/styles.js';
+
+class DialogConfirm extends DialogMixin(LitElement) {
+
+	static get properties() {
+		return {
+			text: { type: String }
+		};
+	}
+
+	static get styles() {
+		return [ dialogStyles, heading3Styles, css`
+			.d2l-dialog-outer {
+				max-width: 420px;
+			}
+			.d2l-dialog-content > div {
+				padding-top: 30px;
+			}
+			.d2l-dialog-header + .d2l-dialog-content > div {
+				padding-top: 0;
+			}
+		`];
+	}
+
+	render() {
+		if (!this._titleId) this._titleId = getUniqueId();
+		if (!this._textId) this._textId = getUniqueId();
+
+		const inner = html`
+			<div class="d2l-dialog-inner">
+				${this.title ? html`
+					<div class="d2l-dialog-header">
+						<div><h2 id="${this._titleId}" class="d2l-heading-3">${this.title}</h2></div>
```

Comment:
Wrapper `<div>` is needed?

Reply:
Yeah.  Basically we have:
```
dialog 
   div (outer - sets up vertical flex container)
      div (header - child flex element in vertical layout)
         div (header layout - sets up horizontal flex container)
            h2
            d2l-button-icon (abort/close)
```
Originally I had a more minimal DOM, but ran into some difficulties.  There were two separate issues:

1. Unable to make `dialog` use `display: flex`. Native dialog seems to rely on `display: none` and `display: block`.  Otherwise the outermost `div` would be eliminated.

2. Unable to have element be both a direct child of a flex container as well as define a new flex container.  This was not consistent across browsers.

There are one or two other issues I ironed out, specifically with Edge and IE11, so I could give try reducing this again.  IE11 and Edge in particular are extra finicky.

Reply:
Ah, gotcha -- that makes sense.